app/src/org/commcare/utils/CommCareUtil.java

```diff
@@ -82,8 +82,7 @@ public static int countPreds(TreeReference reference) {
     }
 
     public static void triggerLogSubmission(Context c) {
-        SharedPreferences settings = CommCareApplication.instance().getCurrentApp().getAppPreferences();
-        String url = settings.getString(CommCareServerPreferences.PREFS_SUBMISSION_URL_KEY, null);
+        String url = LogSubmissionTask.getSubmissionUrl(CommCareApplication.instance().getCurrentApp().getAppPreferences());
```

Comment:
Generally speaking the use of singletons in Java specifically is mostly a way to prevent incredibly annoying and verbose injection of state variables. For us "Application" is more or less persistent across each non-static method call in the app, and if every method had it passed in, things would work normally.

Static methods _wouldn't_ have that dependency, IE: "CommCareApplication._()" wouldn't have existed in their scope before, so to them is an injection of state they didn't have before. It introduces a "secret" dependency that that static method is actually stateful, rather than that it performs some entirely-internal logic (which is what static methods generally can be expected to do).